### PR TITLE
Update the useDirectUpload hook interface 

### PR DESCRIPTION
**Pull request by vbejgrowicz**
## Summary
In order to use useDirectUpload hook for our app we need the following changes:
* allow headers to be set when using the useDirectUpload hook - by configuring them in the ActiveStorageProvider we can use them along with the directUploadsUrl in the hook
* set the `signed_id` in the upload obj on success so that it can be used when sending the attachment to the server - this helps us keep the `signed_id` matched to the correct upload objects
* allow the uploads to be reset 
* allow an upload to be removed 
* styling nit  - switch `uploading` to `isUploading` for readability improvement
* added `successfulUploads` and an `onError` prop to the `useDirectUpload` hook
* added a `DirectUploadResultStatus` enum to use in our app

### CODE REVIEW

File: src/useDirectUpload.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export type Params = {
 }
 
 const useDirectUpload = ({ onSuccess }: Params = {}) => {
-  const { directUploadsUrl } = useConfig();
+  const { directUploadsUrl, headers } = useConfig();
```
Comment (Peleg) [inline src/useDirectUpload.ts:18]: this is probably fine for now, but there's a chance we'll want these to be set an invocation time and not at the config level

File: src/useDirectUpload.ts [hunk 2]
```diff
@@ -38,12 +38,14 @@ const useDirectUpload = ({ onSuccess }: Params = {}) => {
...
-  const uploading = useMemo(() => uploads.some((upload) => upload.status === 'uploading'), [uploads]);
+  const isUploading = useMemo(() => uploads.some((upload) => upload.status === 'uploading'), [uploads]);
 
   return {
     upload,
     uploads,
-    uploading,
+    isUploading,
+    resetUploads: () => setUploads([]),
+    removeUpload: (id: number) => setUploads(uploads.filter(u => u.id !== id))
```
Comment (Peleg) [inline src/useDirectUpload.ts:60]: just curious: what's id here?

**Review by Peleg — APPROVED** [review]

File: src/useDirectUpload.ts (hunk 2 above)
Comment (vbejgrowicz) [reply src/useDirectUpload.ts:60]: id is this taskId that increments for each upload. https://github.com/DogParkLabs/react-native-activestorage/blob/master/src/lib/directUpload.ts#L15

I can switch to doing this by index instead but I liked that it was some sort of identifier.

File: src/useDirectUpload.ts (hunk 1 above)
Comment (vbejgrowicz) [reply src/useDirectUpload.ts:18]: ah okay yeah I figured for these headers it was good that they were in the global config but if we needed different headers per request we can always make the change to add them to the hook or to the upload itself.

**Pull request merged**